### PR TITLE
[AIRFLOW-2794] Add WasbDeleteBlobOperator

### PR DESCRIPTION
**Pull request by bart-eijk**
### Description

* Add `delete_file` method to WasbHook
* Add WasbDeleteBlobOperator

*Additional changes*
The `click` dependency was recently updated to `7.0`, which breaks the build. I have pinpointed it to `6.7` in order to (temporarily) fix this.

### Tests

* Add 4 tests to tests/contrib/hooks/test_wasb_hook.py
* Create new tests in tests/contrib/operators/test_wasb_delete_blob_operator.py

### Code Quality

On my manually patched local installation, running ./scripts/ci/flake8-diff.sh exits without any problems.

Currently, the Travis CI build will fail due to a new version of `pynacl` being released. This is addressed in the following PR by John: https://github.com/apache/incubator-airflow-ci/pull/3

### Jira

https://issues.apache.org/jira/browse/AIRFLOW-2794

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -239,7 +239,7 @@ def write_version(filename=os.path.join(*['airflow',
     + cassandra + mongo
 
 devel = [
-    'click',
+    'click==6.7',
```
Comment (ashb) [inline setup.py]: Sorry to be a pain, but can you make this change a separate PR.

Comment (ashb) [reply setup.py]: Ah NM someone has just created this as a PR. Rebase on to master in a few minutes.

Comment (bart-eijk) [reply setup.py]: Good point. I see that seconds after you posted this kaxil did so: https://github.com/apache/incubator-airflow/pull/3962

Let's wait until this PR is merged, and I'll rebase it on master. :)

**Comment by bart-eijk** [comment]
Rebased on master after `click` was pinned to version 6.7.

**Comment by bart-eijk** [comment]
@ashb The build succeeds now. :-) Could you have a look again, please?

File: airflow/contrib/operators/wasb_delete_blob_operator.py [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+    :param check_options: Optional keyword arguments that
+        `WasbHook.check_for_blob()` takes.
+    :param is_prefix: If blob_name is a prefix, delete all files matching prefix.
+    :type is_prefix: bool
+    :param ignore_if_missing: if True, then return success even if the
+    blob does not exist.
+    :type ignore_if_missing: bool
+    """
+
+    template_fields = ('container_name', 'blob_name', 'wasb_conn_id')
```
Comment (ashb) [inline airflow/contrib/operators/wasb_delete_blob_operator.py]: Templating the connection id is slightly unusually - I don't think any of the other operators do this do they?

Comment (bart-eijk) [reply airflow/contrib/operators/wasb_delete_blob_operator.py]: Good point. None of the other WasbOperators do so.
It does happen in some of the other operators, though, such as:
* airflow.contrib.operators.gcp_container_operator.GKEClusterDeleteOperator
* airflow.contrib.operators.sqoop_operator.SqoopOperator

Do you think we should remove it, so that it's in line with the other WasbOperators?

Comment (bart-eijk) [reply airflow/contrib/operators/wasb_delete_blob_operator.py]: @ashb I have removed 'wasb_conn_id' from the list of template_fields, to make it more in line with the rest of the WasbOperators.

**Comment by bart-eijk** [comment]
@ashb Do you think this is ready to merge? :-)

**Comment by Fokko** [comment]
Thanks @bart-eijk looks good!
